managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -1754,10 +1754,7 @@ synchronized void ledgerClosed(final LedgerHandle lh) {
 
         maybeOffloadInBackground(NULL_OFFLOAD_PROMISE);
 
-        if (!pendingAddEntries.isEmpty()) {
-            // Need to create a new ledger to write pending entries
-            createLedgerAfterClosed();
-        }
+        createLedgerAfterClosed();
```

Comment:
We'd better also remove the other places that call `createLedgerAfterClosed` directly from the ManagedLedgerImpl.java

- rollCurrentLedgerIfFull
- checkInactiveLedgerAndRollOver

The solution introduced a behavior change in the broker. But it looks make sense. All the other places (managed ledger initialization, the task for checking the maximum rollover time ...) are proactively create a new ledger if the current ledger is closed except the ledger rollover by the entry size and entry count. With this solution we can get a better publish latency for the newly published message to a new Ledger (It requires to create a Ledger first and write the message to the Ledger).

If we want to move to lazy ledger creation to reduce the ledgers count. We also need to apply the lazy creation to the managed ledger initialization and other places. It could be different case with this fix.

Reply:
I wonder if we should close the old one and create a new one if the last ledger is empty.

Reply:
BTW, the method `ML.getFirstPosition` will throw  a `NoSuchElementException` if the current ledger was deleted(in other words: `ML.ledgers` is empty).